deepobs/pytorch/testproblems/noise_free_quadratic.py

```diff
@@ -0,0 +1,242 @@
+# -*- coding: utf-8 -*-
+"""TODO Description of the testproblem."""
+
+import scipy.linalg
+import torch
+from torch.utils import data
+
+from ..datasets import dataset
+from .testproblem import UnregularizedTestproblem
+
+
+# The data set
+class data_noise_free_quadratic(dataset.DataSet):
+    """Dataset class for the noise-free quadratic
+
+    The inputs and corresponding labels for training (and similarly for
+    validation and testing) are basically both tensors of size
+    ``train_size`` x ``dim``, where ```dim`` is the dimensionality of the
+    problem. The input data is arbitraty (because it is later multiplied by
+    zero), the labels are zero. The methods below yield the respective
+    data loaders for all three sets.
+    """
+
+    def __init__(
+        self, batch_size, dim, train_size=128, valid_size=128, test_size=128,
+    ):
+        self._dim = dim
+
+        # Check batch size
+        assert batch_size <= min(
+            [train_size, valid_size, test_size]
+        ), "Batch size exceeds size of training/validation/test set"
+
+        self._train_size = train_size
+        self._valid_size = valid_size
+        self._test_size = test_size
+
+        # This attribute is needed by _make_train_eval_dataloader
+        self._train_eval_size = self._train_size
+
+        super(data_noise_free_quadratic, self).__init__(batch_size)
+
+    def _make_train_and_valid_dataloader(self):
+        """Creates the training and validation data loader."""
+
+        # Training data
+        train_data = torch.rand(self._train_size, self._dim)
+        train_labels = torch.zeros(self._train_size, self._dim)
+        train_dataset = data.TensorDataset(train_data, train_labels)
+        train_loader = self._make_dataloader(train_dataset, shuffle=True)
+
+        # Validation data
+        valid_data = torch.rand(self._valid_size, self._dim)
+        valid_labels = torch.zeros(self._valid_size, self._dim)
+        valid_dataset = data.TensorDataset(valid_data, valid_labels)
+        valid_loader = self._make_dataloader(valid_dataset)
+
+        return train_loader, valid_loader
+
+    def _make_test_dataloader(self):
+        """Creates the test data loader."""
+
+        # Test data
+        test_data = torch.rand(self._test_size, self._dim)
+        test_labels = torch.zeros(self._test_size, self._dim)
+        test_dataset = data.TensorDataset(test_data, test_labels)
+        test_loader = self._make_dataloader(test_dataset)
+
+        return test_loader

```

Comment:
Question: Is it necessary to set random seeds here, or does this happen outside these function in DeepOBS?